modules/llrt_child_process/src/lib.rs

```diff
@@ -418,44 +625,204 @@ fn spawn<'js>(
+    let mut stdout = StdioEnum::Piped;
+    let mut stderr = StdioEnum::Piped;
+
+    if let Some(opts) = opts {
+        get_gid(&opts, &mut command)?;
+        get_uid(&opts, &mut command)?;
+        get_cwd(&opts, &mut command)?;
+        get_env(&opts, &mut command)?;
+        get_stdio(&ctx, &opts, &mut stdin, &mut stdout, &mut stderr)?;
+    }
+
+    command.stdin(stdin.to_stdio());
+    command.stdout(stdout.to_stdio());
+    command.stderr(stderr.to_stdio());
+
+    #[cfg(unix)]
+    {
+        command.process_group(0);
+    }
+
+    //tokio command does not have all std command features stabilized
+    let mut command = Command::from(command);
+    ChildProcess::spawn(ctx.clone(), cmd, command_args, command.spawn())
+}
+
+fn exec_file<'js>(
+    ctx: Ctx<'js>,
+    cmd: String,
+    args_and_opts: Rest<Value<'js>>,
+) -> Result<Class<'js, ChildProcess<'js>>> {
+    let args_0 = args_and_opts.first();
+    let args_1 = args_and_opts.get(1);
+    let args_3 = args_and_opts.get(2);
+
+    let cb = get_callback_fn(&ctx, vec![args_0, args_1, args_3])?;
+
+    let mut opts = None;
+    if let Some(arg) = &args_1 {
+        if !arg.is_function() {
+            // is_object() is returning true for array, so checking is_array() aswell
+            if !arg.is_array() && arg.is_object() {
+                opts = arg.as_object().cloned();
+            } else {
+                return Err(Exception::throw_message(
+                    &ctx,
+                    "The \"options\" argument must be of type object.",
+                ));
+            }
+        }
+    }
+
+    let mut command_args = match get_command_args(&ctx, args_0, &mut opts) {
+        Ok(Some(args)) => Some(args),
+        Ok(None) => None,
+        Err(err) => return Err(err),
+    };
+
+    let mut windows_verbatim_arguments: bool = get_windows_verbatim_arguments(opts.as_ref())?;
+    let cmd = get_cmd(
+        opts.as_ref(),
+        &mut command_args,
+        &mut windows_verbatim_arguments,
+        cmd,
+    )?;
+
+    let mut command = StdCommand::new(cmd.clone());
+    set_command_args(
+        &mut command,
+        command_args.as_ref(),
+        windows_verbatim_arguments,
+    );
+
+    let stdin = StdioEnum::Piped;
+    let stdout = StdioEnum::Piped;
+    let stderr = StdioEnum::Piped;
+
+    if let Some(opts) = &opts {
+        get_gid(opts, &mut command)?;
+        get_uid(opts, &mut command)?;
+        get_cwd(opts, &mut command)?;
+        get_env(opts, &mut command)?;
+    }
+
+    command.stdin(stdin.to_stdio());
+    command.stdout(stdout.to_stdio());
+    command.stderr(stderr.to_stdio());
+
+    #[cfg(unix)]
+    {
+        command.process_group(0);
+    }
+
+    //tokio command does not have all std command features stabilized
+    let mut command = Command::from(command);
+    ChildProcess::exec_file(ctx.clone(), cmd, command_args, command.spawn(), cb)
+}
+
+fn get_callback_fn<'js>(
+    ctx: &Ctx<'js>,
+    args: Vec<Option<&Value<'js>>>,
```

Comment:
we should be able to avoid a vec allocation here I am pretty sure `&[Option<&Value<'js>>]` should work